ckan/common.py

```diff
@@ -267,6 +267,28 @@ def aslist(obj: Any, sep: Optional[str] = None, strip: bool = True) -> Any:
         return [obj]
 
 
+def clean_context(
+    context: Context,
+    fields: Optional[List[str]] = None
+) -> Context:
+    """ Copy just the minimum fields into a new context
+        for cases in which we reuse the context and
+        we want a clean version with minimum fields """
+    new_context = {}
+    if fields is None:
+        fields = [
+            'model', 'session', 'user', 'auth_user_obj',
+            'ignore_auth'
+        ]
+
+    for field in fields:
+        if field in context:
+            new_context[field] = context[field]
+
+    new_context = cast(Context, new_context)
+    return new_context
```

Comment:
I like it, but let's keep it simple to make it easier for us to change in the future. If callers want to pass along extra variables they can do so after calling `clean_context`.

```suggestion
def clean_context(
    context: Context,
) -> Context:
    """ Copy just the minimum fields into a new context
        for cases in which we reuse the context and
        we want a clean version with minimum fields """
    new_context = {
        k:context[k] for k in (
            'model', 'session', 'user', 'auth_user_obj', 'ignore_auth'
        ) if k in context
    }
    new_context = cast(Context, new_context)
    return new_context
```